### ISSUE TITLE
Open source server side code and containerize project

### ISSUE THREAD

**Issue by cogito123** [Factorio-Biter-Battles/Factorio-Biter-Battles#506]
It would be nice to have entire factorio server + database ran from docker with trust/quick bar/etc out of box with easy config/setup. In my understanding server side is entirely opaque for non-admin developers which makes it difficult to implement anything that would store persistent states across sessions and integrate new parts that depend on remote compute.

e.g. I have some ideas to implement server-side pathfinder that injects way points into a game at runtime to control NPCs/biters that is otherwise not possible to do in plain scenario code due to performance constrains. By allowing to view existing server side infrastructure - integration of such PRs would be possible

**Comment by Ragnarok77-factorio** [comment]
Yeah agreed, would be nice that it was made public or an equivalent

**Comment by clifffrey** [comment]
I agree that this would be nice.

It will also be a lot of work.  Currently things like database usernames/passwords are sitting in the code (this is possibly OK, as we aren't really relying on the passwords to keep the databases secure, but it is still maybe not a best practice to put that on github). I am also not 100% sure if all of the code authors are comfortable with the code being put up on github (probably they are, but I am not certain).  The other server infrastructure that I have seen is written in a combination of php and node.js, and I am not very familiar with either.

To do all of the work to fully containerize is yet another big chunk of work.

So, I can certainly say that I don't have the time to do this.  If someone is willing to do all of this, and understands the scope of the work involved, I might be willing to help give a few pointers, but there will also be a discussion with neuro and the community about getting access to the server itself to even start this work.

**Comment by uneuro** [comment]
looks like a good idea, but current php code I made is so procedural and made with low skills … I'm afraid it can lead to security (abuse or so) issues or even would need a complete refactor / migration. But at same time it could permit to improve from here. If someone can help me to refactor the code this summer with good practices, the PHP side could go public. About hedwig and JS, need to have the authors ok too.

**Comment by deceptiveSimplicity** [comment]
Something like a docker compose stack?
